src/components/AuthContainer.js

```diff
@@ -1,201 +1,18 @@
-        buttonPrimary({ disabled: this.state.busy, onClick: () => this.register() },
-          'Register'
-        ),
-        this.state.busy && spinner({
-          size: 34, style: { display: null, margin: null, marginLeft: '1ex' }
-        })
-      ])
-    ])
-  }
-
-  async register() {
-    this.setState({ busy: true })
-    await Sam.createUser()
-    const firstName = this.state.givenName
-    const lastName = this.state.familyName
-    const contactEmail = this.state.email
-    await Orchestration.profile.set({ firstName, lastName, contactEmail })
-    this.setState({ busy: false, isRegistered: true })
-  }
-
-  renderSignedOut = () => {
-    return div({ style: { paddingLeft: '1rem', paddingRight: '1rem' } }, [
-      h4('WARNING NOTICE'),
-      p([
-        'You are accessing a US Government web site which may contain information that must be ',
-        'protected under the US Privacy Act or other sensitive information and is intended for ',
-        'Government authorized use only.'
-      ]),
-      p([
-        'Unauthorized attempts to upload information, change information, or use of this web site ',
-        'may result in disciplinary action, civil, and/or criminal penalties. Unauthorized users ',
-        'of this website should have no expectation of privacy regarding any communications or ',
-        'data processed by this website.'
-      ]),
-      p([
-        'Anyone accessing this website expressly consents to monitoring of their actions and all ',
-        'communications or data transiting or stored on related to this website and is advised ',
-        'that if such monitoring reveals possible evidence of criminal activity, NIH may provide ',
-        'that evidence to law enforcement officials.'
-      ]),
-      h4('WARNING NOTICE (when accessing TCGA controlled data)'),
-      p([
-        'You are reminded that when accessing TCGA controlled access information you are bound by ',
-        'the dbGaP TCGA ',
-        link({ target: '_blank', href: 'http://cancergenome.nih.gov/pdfs/Data_Use_Certv082014' }, [
-          'DATA USE CERTIFICATION AGREEMENT (DUCA)'
-        ])
-      ]),
-      hr(),
-      p([
-        'Copyright © 2018, Broad Institute, Inc., Verily Life Sciences LLC'
-      ])
-    ])
-  }
-
-  render() {
-    const { children } = this.props
-    const { isSignedIn, isRegistered } = this.state
-    return h(Fragment, [
-      div({ id: 'signInButton', style: { display: isSignedIn ? 'none' : 'block' } }),
-      Utils.cond(
-        [isSignedIn === undefined, null],
-        [isSignedIn === false, this.renderSignedOut],
-        [isSignedIn === true && isRegistered === undefined, null],
-        [isSignedIn === true && isRegistered === false, this.renderRegistration],
-        children
-      )
-    ])
+export default Utils.connectAtom(authStore, 'authState')(
+  ({ children, authState: { isSignedIn, isRegistered } }) => {
+    return Utils.cond(
+      [isSignedIn === undefined, null],
```

Comment:
`isSignedIn === undefined || isRegistered === undefined`?

Reply:
Yes, that would work.